pyrsa/inference/result.py

```diff
@@ -47,36 +51,82 @@ def __init__(self, models, evaluations, method, cv_method, noise_ceiling):
         self.cv_method = cv_method
         self.noise_ceiling = noise_ceiling
 
-    def save(self, filename):
+    def save(self, filename, file_type='hdf5'):
         """ saves the results into a file. 
+            
+        Args:
+            filename(String): path to the file
+                [or opened file]
+            file_type(String): Type of file to create:
+                hdf5: hdf5 file
+                pkl: pickle file 
 
-        the list of models is currently not saved!
+        """
+        result_dict = self.to_dict()
+        if file_type == 'hdf5':
+            write_dict_hdf5(filename, result_dict)
+        elif file_type == 'pkl':
+            write_dict_pkl(filename, result_dict)
 
-        Args:
-            filename(String): path to the filelocation
+    def to_dict(self):
+        """ Converts the RDMs object into a dict, which can be used for saving
+
+        Returns:
+            results_dict(dict): A dictionary with all the information needed
+                to regenerate the object
 
         """
-        if isinstance(filename, str) and os.path.isfile(filename):
-            raise FileExistsError('File already exists')
-        file = h5py.File(filename, 'a')
-        file['evaluations'] = self.evaluations
-        file['noise_ceiling'] = self.noise_ceiling
-        file.attrs['method'] = self.method
-        file.attrs['cv_method'] = self.cv_method
-        file.attrs['version'] = 3
```

Comment:
We don't want to store the software version anymore?

Reply:
I actually include it again